include/mujincontrollerclient/mujincontrollerclient.h

```diff
@@ -99,6 +106,8 @@ typedef boost::shared_ptr<BinPickingResultResource> BinPickingResultResourcePtr;
 typedef boost::weak_ptr<BinPickingResultResource> BinPickingResultResourceWeakPtr;
 typedef boost::shared_ptr<DebugResource> DebugResourcePtr;
 typedef boost::weak_ptr<DebugResource> DebugResourceWeakPtr;
+typedef boost::shared_ptr<FileEntry> FileEntryPtr;
+typedef boost::weak_ptr<FileEntry> FileEntryWeakPtr;
```

Comment:
Why not `std::vector<FileEntry>`? Why the complicated heap allocation?

Reply:
well, WebResource needs shared_ptr because they are not POD and FileEntry is POD?

now the counterpart is filename std::string...

then because FileEntry does not have destructor?

Reply:
s/counterpart/counterexample/

Reply:
@ziyan ping: Could you explain why WebResource needs shared_ptr while FileEntry is not? FileEntry has non-POD type std::string, so POD stuff does not explain the reason.

Reply:
This structure was proposed in place of `std::pair` you had before. I don't think it is complicated enough to warrant explicit smart pointers, weak pointers. It is a data holder without logic.

Reply:
Can you explain why you think it needs weak pointer?

Reply:
> I don't think it is complicated enough to warrant explicit smart pointers

ok, I'll interpret it as `because FileEntry does not have destructor`.